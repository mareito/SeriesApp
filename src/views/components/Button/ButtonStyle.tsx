import {StyleSheet} from 'react-native';
import {colors} from '../../../globalStyles';

export const styles = StyleSheet.create({
  buttonContainer: {
    width: 200,
    padding: 10,
    borderRadius: 24,
    marginVertical: 10,
  },
  buttonPrimary: {
    backgroundColor: colors.YELLOw,
  },
  buttonSecondary: {
    backgroundColor: colors.WHITE,
  },
  buttonText: {
    alignSelf: 'center',
    fontSize: 18,
    fontWeight: 'bold',
  },
});
