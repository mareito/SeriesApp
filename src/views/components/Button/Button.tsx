import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {colors} from '../../../globalStyles';
import {styles} from './ButtonStyle';

interface ButtonProps {
  text: string;
  variant: 'primary' | 'secondary';
  accion: () => void;
  size: 'regular' | 'small';
}

const Button = ({text, variant, accion, size}: ButtonProps) => {
  return (
    <TouchableOpacity
      onPress={accion}
      style={{
        ...styles.buttonContainer,
        backgroundColor: variant === 'primary' ? colors.YELLOw : colors.WHITE,
        width: size === 'regular' ? 200 : 140,
      }}>
      <View>
        <Text style={styles.buttonText}>{text}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Button;
