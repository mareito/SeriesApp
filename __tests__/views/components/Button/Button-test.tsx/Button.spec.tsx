/**
 * @format
 */

import {render} from '@testing-library/react-native';
import 'react-native';
import Button from '../../../../../src/views/components/Button/Button';

describe('Set pruebas componente Button', () => {
  it('debe renderizar el componente correctamente', () => {
    const component = render(
      <Button text="Ok" variant="primary" accion={() => {}} size="regular" />,
    );
    expect(component).toMatchSnapshot();
    expect(component.getByText('Ok')).toBeTruthy();
  });
});
