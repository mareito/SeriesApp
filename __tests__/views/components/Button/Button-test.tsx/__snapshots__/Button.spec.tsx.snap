// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set pruebas componente Button debe renderizar el componente correctamente 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFD233",
      "borderRadius": 24,
      "marginVertical": 10,
      "opacity": 1,
      "padding": 10,
      "width": 200,
    }
  }
>
  <View>
    <Text
      style={
        {
          "alignSelf": "center",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Ok
    </Text>
  </View>
</View>
`;
